ERCS/erc-X.md

```diff
@@ -112,47 +117,101 @@ Checksum
     - Its extra human-readability will make it more useful as a a stopgap solution until a standard focused on chain and address names is finalized.
 
 ## Appendix A: Binary encoding of CAIP-2 blockchain ID
-The first two bytes are the binary representation of CAIP-2 namespace (see table below), while the remaining bytes correspond to the CAIP-2 reference, whose encoding is namespace-specific and defined below.
+The `ChainType` field is the binary representation of the CAIP namespace (see table below), while the `ChainReference` contains the CAIP-2 reference, whose encoding is namespace-specific and defined below.
 
-### CAIP-2 namespaces' binary representation table
+### CAIP namespaces' binary representation table
 
 | Namespace | binary key |
 | ---       | ---        |
 | eip155    | `0x0000`   |
 | solana    | `0x0002`   |
 
 #### eip155
-The bare `chainid` encoded as a big-endian unsigned integer of the minimum necessary amount of bytes will be used [^1], and leading zeroes will be prohibited. 
+
+##### Text representation
+```
+eip155:<number>
+```
+Where `<number>` is the decimal representation of the chain's `chainid`, without leading zeroes.
+
+##### Binary representation
+The bare `chainid` encoded as a big-endian unsigned integer of the minimum necessary amount of bytes will be used [^1], and leading zeroes will be prohibited.
 
 [^1]: This makes it possible to represent some chains using the full word as their chainid, which CAIP-2 does not support since the set of values representable with 32 `a-zA-Z0-9` characters has less than `type(uint256).max` elements. This is done in an effort to support chains whose ID is the output of `keccak256`, as proposed in ERC-7785.
 
+##### Text -> binary conversion
+Encode the decimal integer as a big-endian unsigned integer using the minimum necessary amount of bytes.
+
+##### Binary -> text conversion
+Compute the decimal representation of the stored big-endian unsigned integer
+
 ##### Examples
-Ethereum Mainnet: `0x000001` (1, encoded as uint8)
+Ethereum Mainnet: `0x01` (1, encoded as uint8)
 
-Optimism Mainnet: `0x00000A` (10, encoded as uint8)
+Optimism Mainnet: `0x0A` (10, encoded as uint8)
 
-Ethereum Sepolia: `0x0000aa36a7` (11155111, encoded as uint24)
+Ethereum Sepolia: `0xaa36a7` (11155111, encoded as uint24)
 
 #### solana
-Solana networks are distinguished by its genesis blockhash, which is to be decoded from base58btc and stored raw in the binary representation and without removing any leading zeroes:
+Solana networks are distinguished by its genesis blockhash, which is normally represented as 44 base58btc characters, corresponding to 32 bytes. We chose to perform the same truncation as CAIP-2[^2] to keep representations trivially convertible between the two, making them also visually identical.
```

Comment:
I think we should be consistent in the decision to truncate ids following CAIP-2 or not...

I'm not fond of truncation, so unless we know of scenarios where it will be necessary to convert CAIP-2 ids to interoperable addresses I think I would avoid truncation altogether.